ArcGISRuntimeSDKQt_QMLSamples/Features/FeatureLayer_Query/FeatureLayer_Query.qml

```diff
@@ -154,13 +126,62 @@ Rectangle {
         }
     }
 
-    // function to form and execute the query
-    function query() {
-        // set the where clause
-        params.whereClause = "STATE_NAME LIKE '" + formatStateNameForQuery(findText.text) + "%'";
+    function selectResults(result, layer) {
+        return new Promise(
+                    (resolve, reject)=>{
+                        // clear any previous selection
+                        layer.clearSelection();
+
+                        // get the features
+                        const features = []
```

Comment:
Is this `while/next` iterator pattern common for us? If we want a copy, can we  do:

```js
const features = Array.from(result.iterator.features);
```

Reply:
This seems like a good pattern to follow, but we need to be careful to test it. `FeatureIterator.features` is a `QQmlListProperty` which does not support forEach. So if that is how `Array.from` is implemented it might not work.

https://bugreports.qt.io/browse/QTBUG-49613

Reply:
that's great! works perfectly!